app/forms/idv/profile_form.rb

```diff
@@ -45,16 +45,28 @@ def initialize_params(params)
       end
     end
 
-    def ssn_signature
-      Pii::Fingerprinter.fingerprint(ssn) if ssn
+    def ssn_signature(key = Pii::Fingerprinter.current_key)
+      Pii::Fingerprinter.fingerprint(ssn, key) if ssn
     end
 
     def ssn_is_unique
       errors.add :ssn, I18n.t('idv.errors.duplicate_ssn') if ssn_is_duplicate?
     end
 
     def ssn_is_duplicate?
-      Profile.where.not(user_id: @user.id).where(ssn_signature: ssn_signature).any?
+      return true if any_matching_ssn_signatures?(Pii::Fingerprinter.current_key)
+      return true if ssn_is_duplicate_with_old_key?
+    end
+
+    def ssn_is_duplicate_with_old_key?
+      KeyRotator::Utils.old_keys(:hmac_fingerprinter_key_queue).each do |key|
```

Comment:
doing an `each` like this to me seems potentially ineffiecient as it's going to make 1 query per old key?

What about an approach that just does all old hashes and 1 query? (I'm hoping/guessing that hashing, while slow, is faster than N+1 database queries?)